app/api/bucketlist_api.py

```diff
@@ -0,0 +1,256 @@
+# Bucketlist resource
+from flask import Flask
+from app.models import db
+from flask_restful import Api, Resource, fields, reqparse
+from flask_jwt import JWT, jwt_required, current_identity
+from app.models.models import User, BucketList, Item
+from flask import request, jsonify, make_response
+from app.api.validate import validate_register, validate_bucketlist, validate_item, validate_limit_and_offset
+
+
+class AddBucketlistResource(Resource):
+    decorators =[jwt_required()]
+    def post(self):
+        '''create new bucketlist'''
+        response = {}
+        json = request.json
+        print(json)
+        validation = validate_bucketlist(json)
+
+        if validation.status:
+            bucketlist = BucketList(name=json['name'], created_by=current_identity.id)
+            db.session.add(bucketlist)
+            db.session.commit()
+            status_code = 201
+        else:
+            status_code = 400
+
+        response["message"] = validation.message
+        response = jsonify(response)
+        response.status_code = status_code
+        return response
+    
+    def get(self):
+        '''List all created bucketlist by name'''
+        response = {}
+        query = request.args.get('q', None) # search string
+        limit = request.args.get('limit', None)
+        offset = request.args.get('offset', None)
+
+        if not (offset or limit):
+            bucketlists = BucketList.query.filter_by(created_by=current_identity.id).all()
+            print(bucketlists)
+            blists = []
+            for blist in bucketlists:
+                blist_object = {
```

Comment:
Since there are multiple places dicts are being created from picking particular attributes from objects, maybe we could create a helper function that'll do that for any object and re-use it in all of the places, e.g
```python
def dict_from(keys, obj):
    return {key: getattr(obj, key) for key in keys}
```
Then use it as
```python
blist_object = dict_from(['id', 'name', 'date_created', 'date_modified'], blist)
```